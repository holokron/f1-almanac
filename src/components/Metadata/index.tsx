import React, { ReactElement } from "react"
import Helmet from "react-helmet"

interface MetadataProps {
  author: string
  description: string
  keywords: string
  title: string
  siteTitle?: string
}

function normalize(item: string | string[]): string[] {
  if (!Array.isArray(item)) {
    item = [item]
  }

  return item.filter(item => item)
}

function Metadata({
  author,
  description,
  keywords,
  title,
  siteTitle = "",
}: MetadataProps): ReactElement<MetadataProps> {
  return (
    <Helmet>
      <title>{normalize([title, siteTitle]).join(" - ")}</title>
      <meta name="description" content={normalize(description).join(" ")} />
      <meta name="keywords" content={normalize(keywords).join(" ")} />
      <meta name="author" content={author} />
    </Helmet>
  )
}

export default React.memo(Metadata)
