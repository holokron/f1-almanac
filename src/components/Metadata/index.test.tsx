import React from "react"
import renderer from "react-test-renderer"
import Metadata from "."

describe("components/Metadata", (): void => {
  it.each([["foo@bar.com", "Foo Bar", "foo bar test", "Foo", "Bar"]])(
    "data: %p",
    (
      author: string,
      description: string,
      keywords: string,
      title: string,
      siteTitle: string = "",
    ): void => {
      const tree = renderer.create(
        <Metadata
          author={author}
          description={description}
          keywords={keywords}
          title={title}
          siteTitle={siteTitle}
        />,
      )

      expect(tree.toJSON()).toMatchSnapshot()
    },
  )
})
