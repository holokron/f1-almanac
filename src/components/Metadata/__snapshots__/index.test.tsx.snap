// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Metadata data: "foo@bar.com" 1`] = `null`;
